Sources/SwiftInspectorVisitors/NestableTypeVisitor.swift

```diff
@@ -88,10 +88,20 @@ public final class NestableTypeVisitor: SyntaxVisitor {
   }
 
   public override func visit(_ node: TypealiasDeclSyntax) -> SyntaxVisitorContinueKind {
-    let typealiasVisitor = TypealiasVisitor(parentType: parentType)
```

Comment:
This line was naive, and didn't follow the pattern of other syntax visitors.

we _need_ to check if we're within a top-level declaration, and create a `newParentType` if we are within a top-level declaration. I've added tests for this logic now.